site/public/js/ta-grading-v2.js

```diff
@@ -596,26 +785,46 @@ function updateTwoPanelLayout () {
     }
   }
   // finally append the latest panels to their respective buckets
-  if (leftPanel) {
-    document.querySelector(taLayoutDet.panelsBucket.leftSelector).append(leftPanel);
+  if (leftTopPanel) {
+    document.querySelector(taLayoutDet.panelsBucket.leftTopSelector).append(leftTopPanel);
+  }
+  if (leftBottomPanel) {
+    document.querySelector(taLayoutDet.panelsBucket.leftBottomSelector).append(leftBottomPanel);
   }
-  if (rightPanel) {
-    document.querySelector(taLayoutDet.panelsBucket.rightSelector).append(rightPanel);
+  if (rightTopPanel) {
+    document.querySelector(taLayoutDet.panelsBucket.rightTopSelector).append(rightTopPanel);
+  }
+  if (rightBottomPanel) {
+    document.querySelector(taLayoutDet.panelsBucket.rightBottomSelector).append(rightBottomPanel);
   }
   saveTaLayoutDetails();
 }
 
 // Exchanges positions of left and right panels
 function exchangeTwoPanels () {
-  if (taLayoutDet.currentTwoPanels.left && taLayoutDet.currentTwoPanels.right) {
-    const leftPanel = taLayoutDet.currentTwoPanels.left;
+  if (+taLayoutDet.numOfPanelsEnabled === 2) {
     taLayoutDet.currentTwoPanels = {
-      left: taLayoutDet.currentTwoPanels.right,
-      right: leftPanel
+      leftTop: taLayoutDet.currentTwoPanels.rightTop,
+      rightTop: taLayoutDet.currentTwoPanels.leftTop,
     };
-    updateTwoPanelLayout();
-  } else {
-      alert("Exchange works only when there are two panels...");
+    updatePanelLayoutModes();
+  }
+  else if (+taLayoutDet.numOfPanelsEnabled === 3) {
+    taLayoutDet.currentTwoPanels = {
+      leftTop: taLayoutDet.currentTwoPanels.rightTop,
+      leftBottom: taLayoutDet.currentTwoPanels.rightBottom,
+      rightTop: taLayoutDet.currentTwoPanels.leftTop,
+      rightBottom: taLayoutDet.currentTwoPanels.leftBottom,
+    };
+    $(".panel-item-section.left-bottom, .panel-item-section.right-bottom, .panel-item-section-drag-bar").toggleClass("active");
+    taLayoutDet.dividedColName = $(".panel-item-section.right-bottom").is(":visible") ? "RIGHT" : "LEFT";
+    updatePanelOptions();
+    updatePanelLayoutModes();
+    initializeHorizontalTwoPanelDrag();
+  }
+  else {
+    // taLayoutDet.numOfPanelsEnabled is 1
+    alert("Exchange works only when there are two panels...");
```

Comment:
I don't think this error message makes sense in this context, something like "Cannot swap left and right panels outside of two pannel mode" would be more descriptive